### ISSUE TITLE
Pip installable (z5py as optional dependency)

### ISSUE THREAD

**Issue by clbarnes** [constantinpape/heimdall#3]
Really looking forward to seeing this develop! If at all possible, (eventually) using this outside of conda would be really helpful (if this library is going to stay pure-python), even if that meant having z5py as an optional dependency and forgoing n5/zarr compatibility (there are plenty of other array-likes in the world!). No rush at all, I know you're not onto packaging yet, I just wanted to mention it now because considering that use case in the early stages may be easier than trying to excise conda dependencies at a later date.

**Comment by constantinpape** [comment]
> If at all possible, (eventually) using this outside of conda would be really helpful (if this library is going to stay pure-python), even if that meant having z5py as an optional dependency and forgoing n5/zarr compatibility (there are plenty of other array-likes in the world!)

Yes, I plan to make this available on pip as well and this repo will most likely be pure python.
I also want to use `zarr-python` (which is available via pip) as a fall-back to `z5py`, so that there is at least zarr array support without python (I think `n5` support in `zarr-python` is not fully there yet).

I realized yesterday that a lot of functionality / convenience functions that would be very handy here (and also in general) is crammed into my [cluster_tools/utils](https://github.com/constantinpape/cluster_tools/tree/master/cluster_tools/utils), which needs some nasty dependencies.
So my next step will be to refactor this into a stand-alone repo and after that I will continue here.
